onnxruntime/test/shared_lib/test_nontensor_types.cc

```diff
@@ -28,199 +28,110 @@ struct RelAllocations {
+    auto temp = seq_ort.GetValue(999, default_allocator.get());
+  } catch (const Ort::Exception& e) {
+    failed = e.GetOrtErrorCode() == ORT_RUNTIME_EXCEPTION;
+  }
+  ASSERT_EQ(failed, true);
 
   // Fetch
   for (int idx = 0; idx < N; ++idx) {
-    OrtValue* map_out = nullptr;
-    OrtStatus* st = OrtGetValue(seq_ort, idx, default_allocator.get(), &map_out);
-    rel.add(map_out);
-    rels.add(st);
-    ASSERT_EQ(st, nullptr);
-    ASSERT_NE(map_out, nullptr);
+    Ort::Value map_out = seq_ort.GetValue(idx, default_allocator.get());
 
     // fetch the map
     // first fetch the keys
-    OrtValue* keys_ort = nullptr;
-    st = OrtGetValue(map_out, 0, default_allocator.get(), &keys_ort);
-    rel.add(keys_ort);
-    rels.add(st);
-    ASSERT_EQ(st, nullptr);
-    ASSERT_NE(keys_ort, nullptr);
-
-    std::unique_ptr<int64_t> keys_ret_u;
-    int64_t* keys_ret = keys_ret_u.get();
-    st = OrtGetTensorMutableData(keys_ort, reinterpret_cast<void**>(&keys_ret));
-    rels.add(st);
-    ASSERT_EQ(st, nullptr);
-    ASSERT_NE(keys_ret, nullptr);
+    Ort::Value keys_ort = map_out.GetValue(0, default_allocator.get());
+
+    int64_t* keys_ret = keys_ort.GetTensorMutableData<int64_t>();
     ASSERT_EQ(std::set<int64_t>(keys_ret, keys_ret + NUM_KV_PAIRS),
               std::set<int64_t>(std::begin(keys), std::end(keys)));
 
     // second fetch the values
-    OrtValue* values_ort = nullptr;
-    st = OrtGetValue(map_out, 1, default_allocator.get(), &values_ort);
-    rel.add(values_ort);
-    rels.add(st);
-    ASSERT_EQ(st, nullptr);
-    ASSERT_NE(values_ort, nullptr);
-
-    std::unique_ptr<float> values_ret_u;
-    float* values_ret = values_ret_u.get();
-    st = OrtGetTensorMutableData(values_ort, reinterpret_cast<void**>(&values_ret));
-    rels.add(st);
-    ASSERT_EQ(st, nullptr);
-    ASSERT_NE(values_ret, nullptr);
+    Ort::Value values_ort = map_out.GetValue(1, default_allocator.get());
+
+    float* values_ret = values_ort.GetTensorMutableData<float>();
     ASSERT_EQ(std::set<float>(values_ret, values_ret + NUM_KV_PAIRS),
               std::set<float>(std::begin(values), std::end(values)));
   }
 }
 
 TEST_F(CApiTest, CreateGetVectorOfMapsStringFloat) {  // support zipmap output type seq(map(string, float))
   // Creation
-  std::unique_ptr<MockedOrtAllocator> default_allocator(std::make_unique<MockedOrtAllocator>());
-  OrtAllocatorInfo* info;
-  ORT_THROW_ON_ERROR(OrtCreateAllocatorInfo("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault, &info));
-  std::unique_ptr<OrtAllocatorInfo, decltype(&OrtReleaseAllocatorInfo)> rel_info(info, OrtReleaseAllocatorInfo);
-
-  RelAllocations<OrtValue> rel(&OrtReleaseValue);
-  RelAllocations<OrtStatus> rels(&OrtReleaseStatus);
+  auto default_allocator = std::make_unique<MockedOrtAllocator>();
+  Ort::AllocatorInfo info("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault);
 
   const int N = 3;
   const int64_t NUM_KV_PAIRS = 4;
-  std::vector<OrtValue*> in(N);
+  std::vector<Ort::Value> in;
```

Comment:
`in(N)` ?

Reply:
Ort::Value has no default constructor (intentionally) so this isn't possible.